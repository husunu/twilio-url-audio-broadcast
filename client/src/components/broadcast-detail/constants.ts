import { Broadcast, Channel } from '@/types';

export const defaultBroadcast: Broadcast = {
  channel: Channel.VOICE,
  friendlyName: '',
  from: '',
  broadcastId: '',
  message: '',
  dateCreated: new Date(),
  canceled: false,
  completed: false,
};

export const LOADER_TEST_ID = 'broadcast-detail-loader';
export const RECIPIENTS_TABLE_TEST_ID = 'recipients-table';
export const PAGINATOR_TEST_ID = 'paginator';
