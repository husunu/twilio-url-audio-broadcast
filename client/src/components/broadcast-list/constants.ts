export const LOADER_TEST_ID = 'broadcast-list-loader';
export const REQUIREMENTS = [
  'Active Twilio Phone number with Voice capabilities',
  'List of recipient phone numbers (.txt)',
  'Message you want to send',
];
