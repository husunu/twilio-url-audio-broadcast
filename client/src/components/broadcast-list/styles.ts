import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { green, grey, orange } from '@material-ui/core/colors';
import { CallStatus } from '@/types';

type Possiblestatus =
  | CallStatus.COMPLETED
  | CallStatus.IN_PROGRESS
  | CallStatus.CANCELED;

const colors: { [key in Possiblestatus]: string } = {
  [CallStatus.CANCELED]: orange[700],
  [CallStatus.IN_PROGRESS]: grey[700],
  [CallStatus.COMPLETED]: green[700],
};

export const notificationItemStyles = makeStyles((theme: Theme) =>
  createStyles({
    listItem: {
      marginTop: '16px',
      backgroundColor: theme.palette.grey[50],
      '&:hover': {
        textDecoration: 'none',
        backgroundColor: theme.palette.grey[100],
      },
    },
    notificationStatus: (props: { status: Possiblestatus }) => ({
      color: colors[props.status],
      fontWeight: 'bold',
      textTransform: 'capitalize',
    }),
  })
);

export default makeStyles((theme: Theme) =>
  createStyles({
    root: {
      position: 'relative',
      display: 'flex',
      width: '90%',
      margin: '0 auto',
      padding: '2rem',
      flexDirection: 'column',
      flexWrap: 'nowrap',
      justifyContent: 'flex-start',
    },
    subtitle: {
      margin: '32px 0',
    },
    requirements: {
      paddingInlineStart: '2rem',
    },
    createButton: {
      marginTop: '5px',
    },
  })
);
