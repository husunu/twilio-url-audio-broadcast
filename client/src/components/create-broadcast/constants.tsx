import React from 'react';

import { StepItem } from './types';
import { StepType } from '@/types';

import { Configure, RecipientList, Review } from './steps';

import {
  Settings as SettingsIcon,
  GroupAdd as GroupAddIcon,
  Send as SendIcon,
} from '@material-ui/icons';

const locations = {
  configure: '/create/configure',
  recipients: '/create/recipients',
  review: '/create/review',
};

export const steps: Array<StepItem> = [
  {
    key: StepType.Configure,
    label: 'Configure Notification',
    icon: <SettingsIcon />,
    location: locations.configure,
    component: Configure,
    next: locations.recipients,
  },
  {
    key: StepType.Recipients,
    label: 'Upload Recipients',
    icon: <GroupAddIcon />,
    location: locations.recipients,
    component: RecipientList,
    prev: locations.configure,
    next: locations.review,
  },
  {
    key: StepType.Review,
    label: 'Review & Send',
    icon: <SendIcon />,
    location: locations.review,
    component: Review,
    prev: locations.recipients,
    next: '',
  },
];
