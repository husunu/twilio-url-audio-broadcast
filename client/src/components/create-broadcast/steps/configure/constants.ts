export const buyNewNumberUrl =
  'https://www.twilio.com/console/phone-numbers/search';
