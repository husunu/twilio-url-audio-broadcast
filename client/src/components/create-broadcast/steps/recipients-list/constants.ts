export const MAX_NUMBERS = 500;

export const ALERTS = {
  FILE_READ_ERROR: 'There was an error reading the file.',
  FILE_EXCEEDS_MAX: `File has more than ${MAX_NUMBERS} numbers. Please review issues and upload again.`,
  INVALID_NUMBERS:
    'The list of recipients has invalid numbers. Please review issues and upload again.',
  PARSE_SUCCESS: 'The list of recipients is successfully loaded.',
  FILE_REJECT_ERROR: 'Please make sure you tried to upload a text file (.txt)',
};

export const INVALID_NUMBERS = 'invalidNumbers';
export const VALID_NUMBERS = 'validNumbers';

export const INITIAL_VALIDATION_RESULT = {
  [INVALID_NUMBERS]: [],
  [VALID_NUMBERS]: [],
};
