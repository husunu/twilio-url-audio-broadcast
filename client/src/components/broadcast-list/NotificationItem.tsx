import { notificationItemStyles as useStyles } from './styles';
import { Box, Button, Paper, Typography } from '@material-ui/core';
import React, { useMemo } from 'react';
import { CallStatus } from '@/types';

type NotificationItemType = {
  broadcastId: string;
  friendlyName: string;
  dateCreated: Date;
  completed: boolean;
  canceled: boolean;
};

export const NotificationItem = ({
  broadcastId,
  friendlyName,
  dateCreated,
  completed,
  canceled,
}: NotificationItemType) => {
  const status = useMemo(() => {
    return canceled
      ? CallStatus.CANCELED
      : completed
      ? CallStatus.COMPLETED
      : CallStatus.IN_PROGRESS;
  }, [canceled, completed]);

  const classes = useStyles({ status });

  return (
    <Paper elevation={2} key={broadcastId} className={classes.listItem}>
      <Box display="flex" flexDirection="row" p={2}>
        <Box flexGrow="1" display="flex" flexDirection="column">
          <Box fontSize={18} fontWeight={500}>
            {friendlyName}
          </Box>
          <Typography variant="caption">
            {new Date(dateCreated).toLocaleDateString()} Î‡{' '}
            <span className={classes.notificationStatus}>{status}</span>
          </Typography>
        </Box>
        <Button
          variant="outlined"
          color="primary"
          href={`/broadcasts/${broadcastId}`}
        >
          VIEW REPORT
        </Button>
      </Box>
    </Paper>
  );
};
