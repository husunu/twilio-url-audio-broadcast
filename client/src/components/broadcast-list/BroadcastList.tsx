import React, { useCallback, useState } from 'react';
import Pagination from '@material-ui/lab/Pagination';
import useFetch from 'use-http';
import useStyles from './styles';
import { Add } from '@material-ui/icons';
import { Link as RouterLink } from 'react-router-dom';
import { GreyContainer } from '@/components/common/containers';
import { NotificationItem } from './NotificationItem';
import { StyledCard } from '@/components/common/styled-card';
import { Broadcast } from '@/types';
import { Alert, AlertType } from '@/components/common/alert';
import { Box, Button, CircularProgress, Typography } from '@material-ui/core';
import { LOADER_TEST_ID, REQUIREMENTS } from './constants';

const alert = {
  message:
    'Sorry, we could not retrieve your voice notifications history. Please try again',
  type: AlertType.ERROR,
};

export const BroadcastList: React.FC = () => {
  const [currentPage, setCurrentPage] = useState(1);
  const classes = useStyles();

  const { data, loading, error } = useFetch(
    `/api/broadcasts?page=${currentPage - 1}`,
    {
      method: 'GET',
      data: { broadcasts: [], pageCount: 0 },
    },
    [currentPage]
  );

  const {
    broadcasts,
    pageCount,
  }: { broadcasts: Broadcast[]; pageCount: number } = data;

  const onPaginationChange = useCallback(
    (event: React.ChangeEvent<unknown>, value: number) => {
      setCurrentPage(value);
    },
    [setCurrentPage]
  );

  return (
    <StyledCard className={classes.root}>
      <Box display="flex" flexDirection="row">
        <Box flexGrow="1">
          <Typography variant="h4" component="h1">
            My Voice Notifications
          </Typography>
          <Typography className={classes.subtitle}>
            Review all of your voice notifications and access reports
          </Typography>
        </Box>
        <Box>
          <Button
            component={RouterLink}
            to="/create"
            color="primary"
            variant="contained"
            className={classes.createButton}
            startIcon={<Add />}
          >
            New Voice Notification
          </Button>
        </Box>
      </Box>
      {loading && (
        <Box data-testid={LOADER_TEST_ID}>
          <CircularProgress />
        </Box>
      )}
      {error && <Alert alert={alert} showReload={true} />}
      {!error && !loading && broadcasts?.length > 0 && (
        <>
          {broadcasts.map(
            ({
              broadcastId,
              friendlyName,
              dateCreated,
              canceled,
              completed,
            }) => (
              <NotificationItem
                key={broadcastId}
                friendlyName={friendlyName}
                broadcastId={broadcastId}
                dateCreated={dateCreated}
                canceled={canceled}
                completed={completed}
              />
            )
          )}
          {pageCount > 1 && (
            <Box
              margin={4}
              display="flex"
              flexDirection="row"
              justifyContent="center"
            >
              <Pagination
                count={pageCount}
                page={currentPage}
                onChange={onPaginationChange}
                color="primary"
              />
            </Box>
          )}
        </>
      )}
      {!error && !loading && broadcasts?.length === 0 && (
        <GreyContainer>
          You haven't created any voice notification yet. You can create your
          first voice notification in 3 easy steps, only be sure to have the
          following
          <ul>
            {REQUIREMENTS.map((requirement, i) => (
              <li key={i}>{requirement}</li>
            ))}
          </ul>
        </GreyContainer>
      )}
    </StyledCard>
  );
};
