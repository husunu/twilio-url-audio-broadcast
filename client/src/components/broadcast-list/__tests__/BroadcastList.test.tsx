import React from 'react';
import { waitFor, cleanup } from '@testing-library/react';
import { FetchMock } from 'jest-fetch-mock';
import moment from 'moment-timezone';

import { BroadcastList } from '@/components';
import { renderWithAppContexts } from '@/testUtils';
import { LOADER_TEST_ID } from '@/components/broadcast-list/constants';

const fetch = global.fetch as FetchMock;
moment.tz.setDefault('America/Los_Angeles');

describe('BroadcastList', () => {
  const emptyBroadcastListResponse = { broadcasts: [] };
  const sampleBroadcastListResponse = {
    broadcasts: [
      {
        broadcastId: '1',
        friendlyName: 'John Doe',
        dateCreated: moment('2020-05-04').toISOString(),
      },
      {
        broadcastId: '2',
        friendlyName: 'Jane Doe',
        dateCreated: moment('2020-01-01').toISOString(),
      },
      {
        broadcastId: '3',
        friendlyName: 'Foo Bar',
        dateCreated: moment('2019-11-11').toISOString(),
      },
    ],
    currentPage: 0,
    pageCount: 4,
  };

  afterEach(() => {
    cleanup();
    fetch.resetMocks();
  });

  test('loader is shown while loading', async () => {
    fetch.mockResponse(
      () =>
        new Promise((resolve, reject) =>
          setTimeout(() => reject({ body: 'ko' }), 100)
        )
    );
    const { getByTestId } = renderWithAppContexts(<BroadcastList />);
    expect(getByTestId(LOADER_TEST_ID)).toBeInTheDocument();
  });

  test('when request is successful broadcast list is rendered correctly', async () => {
    fetch.mockResponse(JSON.stringify(sampleBroadcastListResponse), {
      status: 200,
    });
    const { container } = renderWithAppContexts(<BroadcastList />);
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  test('when request is unsuccessful an appropriate feedback message is shown', async () => {
    fetch.mockReject(new Error('Internal Server Error'));
    const { getByText } = renderWithAppContexts(<BroadcastList />);
    await waitFor(() => {
      expect(
        getByText(
          /Sorry, we could not retrieve your voice notifications history. Please try again/i
        )
      ).toBeInTheDocument();
    });
  });

  test('when there are no broadcasts an appropriate feedback message is shown', async () => {
    fetch.mockResponse(JSON.stringify(emptyBroadcastListResponse), {
      status: 200,
    });
    const { getByText } = renderWithAppContexts(<BroadcastList />);
    await waitFor(() => {
      expect(
        getByText(
          /You can create your first voice notification in 3 easy steps/i
        )
      ).toBeInTheDocument();
    });
  });
});
