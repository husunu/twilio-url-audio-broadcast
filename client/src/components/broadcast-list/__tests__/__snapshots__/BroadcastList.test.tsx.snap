// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BroadcastList when request is successful broadcast list is rendered correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root-138 WithStyles(ForwardRef(Card))-root-139 makeStyles-root-134 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-170"
    >
      <div
        class="MuiBox-root MuiBox-root-171"
      >
        <h1
          class="MuiTypography-root MuiTypography-h4"
        >
          My Voice Notifications
        </h1>
        <p
          class="MuiTypography-root makeStyles-subtitle-135 MuiTypography-body1"
        >
          Review all of your voice notifications and access reports
        </p>
      </div>
      <div
        class="MuiBox-root MuiBox-root-202"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-createButton-137 MuiButton-containedPrimary"
          href="/create"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            New Voice Notification
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-244"
      data-testid="broadcast-list-loader"
    >
      <div
        class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`BroadcastList when request is successful broadcast list is rendered correctly 2`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root-138 WithStyles(ForwardRef(Card))-root-139 makeStyles-root-134 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-170"
    >
      <div
        class="MuiBox-root MuiBox-root-171"
      >
        <h1
          class="MuiTypography-root MuiTypography-h4"
        >
          My Voice Notifications
        </h1>
        <p
          class="MuiTypography-root makeStyles-subtitle-135 MuiTypography-body1"
        >
          Review all of your voice notifications and access reports
        </p>
      </div>
      <div
        class="MuiBox-root MuiBox-root-202"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-createButton-137 MuiButton-containedPrimary"
          href="/create"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            New Voice Notification
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-listItem-267 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-270"
      >
        <div
          class="MuiBox-root MuiBox-root-271"
        >
          <div
            class="MuiBox-root MuiBox-root-272"
          >
            John Doe
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            5/4/2020
             ·
             
            <span
              class="makeStyles-notificationStatus-268 makeStyles-notificationStatus-269"
            >
              in-progress
            </span>
          </span>
        </div>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          href="/broadcasts/1"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            VIEW REPORT
          </span>
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-listItem-267 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-274"
      >
        <div
          class="MuiBox-root MuiBox-root-275"
        >
          <div
            class="MuiBox-root MuiBox-root-276"
          >
            Jane Doe
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            1/1/2020
             ·
             
            <span
              class="makeStyles-notificationStatus-268 makeStyles-notificationStatus-273"
            >
              in-progress
            </span>
          </span>
        </div>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          href="/broadcasts/2"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            VIEW REPORT
          </span>
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-listItem-267 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-278"
      >
        <div
          class="MuiBox-root MuiBox-root-279"
        >
          <div
            class="MuiBox-root MuiBox-root-280"
          >
            Foo Bar
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            11/11/2019
             ·
             
            <span
              class="makeStyles-notificationStatus-268 makeStyles-notificationStatus-277"
            >
              in-progress
            </span>
          </span>
        </div>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          href="/broadcasts/3"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            VIEW REPORT
          </span>
        </a>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-281"
    >
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiPaginationItem-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-selected"
              tabindex="0"
              type="button"
            >
              1
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 2"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
              tabindex="0"
              type="button"
            >
              2
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 3"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
              tabindex="0"
              type="button"
            >
              3
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 4"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
              tabindex="0"
              type="button"
            >
              4
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiPaginationItem-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;

exports[`BroadcastList when request is successful broadcast list is rendered correctly 3`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root WithStyles(ForwardRef(Card))-root-138 WithStyles(ForwardRef(Card))-root-139 makeStyles-root-134 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-170"
    >
      <div
        class="MuiBox-root MuiBox-root-171"
      >
        <h1
          class="MuiTypography-root MuiTypography-h4"
        >
          My Voice Notifications
        </h1>
        <p
          class="MuiTypography-root makeStyles-subtitle-135 MuiTypography-body1"
        >
          Review all of your voice notifications and access reports
        </p>
      </div>
      <div
        class="MuiBox-root MuiBox-root-202"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-createButton-137 MuiButton-containedPrimary"
          href="/create"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </span>
            New Voice Notification
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-listItem-267 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-270"
      >
        <div
          class="MuiBox-root MuiBox-root-271"
        >
          <div
            class="MuiBox-root MuiBox-root-272"
          >
            John Doe
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            5/4/2020
             ·
             
            <span
              class="makeStyles-notificationStatus-268 makeStyles-notificationStatus-269"
            >
              in-progress
            </span>
          </span>
        </div>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          href="/broadcasts/1"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            VIEW REPORT
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-listItem-267 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-274"
      >
        <div
          class="MuiBox-root MuiBox-root-275"
        >
          <div
            class="MuiBox-root MuiBox-root-276"
          >
            Jane Doe
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            1/1/2020
             ·
             
            <span
              class="makeStyles-notificationStatus-268 makeStyles-notificationStatus-273"
            >
              in-progress
            </span>
          </span>
        </div>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          href="/broadcasts/2"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            VIEW REPORT
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiPaper-root makeStyles-listItem-267 MuiPaper-elevation2 MuiPaper-rounded"
    >
      <div
        class="MuiBox-root MuiBox-root-278"
      >
        <div
          class="MuiBox-root MuiBox-root-279"
        >
          <div
            class="MuiBox-root MuiBox-root-280"
          >
            Foo Bar
          </div>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            11/11/2019
             ·
             
            <span
              class="makeStyles-notificationStatus-268 makeStyles-notificationStatus-277"
            >
              in-progress
            </span>
          </span>
        </div>
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          href="/broadcasts/3"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            VIEW REPORT
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-281"
    >
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiPaginationItem-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary Mui-selected"
              tabindex="0"
              type="button"
            >
              1
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 2"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
              tabindex="0"
              type="button"
            >
              2
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 3"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
              tabindex="0"
              type="button"
            >
              3
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 4"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
              tabindex="0"
              type="button"
            >
              4
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-textPrimary"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiPaginationItem-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
